codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/AwsHttpPresignURLClientGenerator.java

```diff
@@ -204,6 +212,8 @@ public void writeAdditionalFiles(
             // generate client helpers such as copyAPIClient, GetAPIClientOptions()
             writePresignClientHelpers(writer, model, symbolProvider, serviceShape);
 
+            writer.write(new PresignContextPolyfillMiddleware(serviceShape));
+
```

Comment:
nit: I'm not too familiar with Go conventions, but is using the term polyfill idiomatic?

Reply:
Guess it is now? The resulting middleware is also unexported fwiw.